packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,147 @@
+import { execFile as execFileCb } from 'child_process';
+import * as path from 'path';
+import { promisify } from 'util';
+import { buildEsbuildBrowserInternal } from '../browser-esbuild';
+import { BrowserEsbuildOptions } from '../browser-esbuild/options';
+import { OutputHashing } from '../browser-esbuild/schema';
+import { normalizeOptions } from './options';
+import { Schema as JestBuilderSchema } from './schema';
+import { findTestFiles } from './test-files';
+
+const execFile = promisify(execFileCb);
+
+/** Main execution function for the Jest builder. */
+export default createBuilder(
+  async (schema: JestBuilderSchema, context: BuilderContext): Promise<BuilderOutput> => {
+    context.logger.warn(
+      'NOTE: The Jest builder is currently EXPERIMENTAL and not ready for production use.',
+    );
+
+    const options = normalizeOptions(schema);
+    const testOut = 'dist/test-out'; // TODO(dgp1130): Hide in temp directory.
+
+    // Verify Jest installation and get the path to it's binary.
+    // We need to `node_modules/.bin/jest`, but there is no means to resolve that directly. Fortunately Jest's `package.json` exports the
+    // same file at `bin/jest`, so we can just resolve that instead.
+    const jest = require.resolve('jest/bin/jest');
+    if (!jest) {
+      return {
+        success: false,
+        error:
+          'Jest is not installed, most likely you need to run `npm install jest --save-dev` in your project.',
+      };
+    }
+
+    // Verify that JSDom is installed in the project.
+    const environment = require.resolve('jest-environment-jsdom');
+    if (!environment) {
+      return {
+        success: false,
+        error:
+          '`jest-environment-jsdom` is not installed. Install it with `npm install jest-environment-jsdom --save-dev`.',
+      };
+    }
+
+    // Build all the test files.
+    const testFiles = await findTestFiles(options, context.workspaceRoot);
+    const jestGlobal = path.join(__dirname, 'jest-global.mjs');
+    const initTestBed = path.join(__dirname, 'init-test-bed.mjs');
+    const buildResult = await build(context, {
+      // Build all the test files and also the `jest-global` and `init-test-bed` scripts.
+      entryPoints: new Set(Array.from(testFiles).concat([jestGlobal, initTestBed])),
+      tsConfig: options.tsConfig,
+      polyfills: options.polyfills ?? ['zone.js', 'zone.js/testing'],
+      outputPath: testOut,
+      aot: false,
+      index: null,
+      outputHashing: OutputHashing.None,
+      outExtension: 'mjs', // Force native ESM.
+      commonChunk: false,
+      optimization: false,
+      buildOptimizer: false,
+      sourceMap: {
+        scripts: true,
+        styles: false,
+        vendor: true,
+      },
+    });
+    if (!buildResult.success) {
+      return buildResult;
+    }
+
+    // Execute Jest on the built output directory.
+    const jestProc = execFile('node', [
```

Comment:
I believe using `spawn` is a better here.

From my understanding `execFile` will cause the data to be buffered once the processes closes, this might cause users to think that the tests hanged as they don't get any feedback until all tests have executed. Unlike spawn where there is a continues data flow.